### PR TITLE
if storage is specified create volume and attach to machine

### PR DESCRIPTION
**Pull request by jazzdan**

### CODE REVIEW

**Comment by jazzdan** [comment]
Current dependencies on/for this PR:
* main
  * **PR #5** <a href="https://app.graphite.dev/github/pr/replayio/fly-buildkite-plugin/5" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/replayio/fly-buildkite-plugin/5?utm_source=stack-comment).

File: fly/proxy.ts [hunk 1]
```diff
@@ -112,7 +171,28 @@ export class FlyProxy {
...
+        return this.startMachineInner(
+          namePrefix,
+          image,
+          cpus,
+          memory,
+          storageInGB,
+          env,
+          attempts + 1,
+          // remove the region we just tried from the list
+          regionsToTry.filter((r) => r !== region),
```
Comment (loganfsmyth) [inline fly/proxy.ts:189]: Why would this fail in certain regions?

Comment (jazzdan) [reply fly/proxy.ts:189]: Sometimes fly has capacity issues. This actually hasn't happened in a while but over the summer it was pretty frequent.

Comment (loganfsmyth) [reply fly/proxy.ts:189]: Ah got it

**Review by loganfsmyth — APPROVED** [review]

**Review by loganfsmyth — APPROVED** [review]

**Pull request merged**